File: src/main/java/ru/spbstu/spartamonitor/listeners/ParserListener.java
package ru.spbstu.spartamonitor.listeners;

import com.google.common.eventbus.Subscribe;
import ru.spbstu.spartamonitor.SpartaMonitorController;
import ru.spbstu.spartamonitor.events.ParserEvent;

public record ParserListener(SpartaMonitorController controller) {
    @Subscribe
    public void handleParserEvent(ParserEvent event) {
        controller.currentFrameNumber.setText(String.format("%d", event.countFrames()));
    }
}

File: src/main/kotlin/ru/spbstu/spartamonitor/listeners/ParserListener.kt
package ru.spbstu.spartamonitor.listeners

import com.google.common.eventbus.Subscribe
import ru.spbstu.spartamonitor.SpartaMonitorController
import ru.spbstu.spartamonitor.events.ParserEvent

data class ParserListener(val controller: SpartaMonitorController) {
    @Subscribe
    fun handleParserEvent(event: ParserEvent) {
        controller.currentFrameNumber.text = event.countFrames.toString()
    }
}
